Unlock party fixes for Haschel leaving after Gehrich fight
*Just in case they get read elsewhere, s24 and s25 are kept updated. Might be safe to delete.
--- original
+++ modified
@@ -1246,6 +1246,7 @@
 call 103, stor[24], 0x0, 0x0, 0x0
 jmp inl[:LABEL_56]
 LABEL_148:
# Check slot 0 for Haschel
+jmp_cmp ==, 0x4, var[17][0], inl[:HASCHEL_SLOT0]
 mov var[17][1], stor[24]
 mov var[17][2], stor[25]
 jmp_cmp !=, 0x4, stor[24], inl[:LABEL_149]
@@ -1257,20 +1258,26 @@
 LABEL_150:
 mov 0x6, stor[25]
 jmp_cmp !=, stor[25], stor[24], inl[:LABEL_151]
# Check slot 0 for Shana
+jmp_cmp ==, 0x2, var[17][0], inl[:EMPTY_SLOT2]
 mov 0x2, stor[25]
 LABEL_151:
# Check slot 0 for Meru
+jmp_cmp ==, 0x6, var[17][0], inl[:EMPTY_SLOT2]
 mov stor[25], var[17][2]
 LABEL_152:
# Remove unnecessary flag resets
-andor 0x40, 0x23, var[115 + 0]
-andor 0x40, 0x0, var[115 + 1]
-andor 0x40, 0x3, var[115 + 2]
-andor 0x40, 0x3, var[115 + 3]
 andor 0x40, 0x0, var[115 + 4]
-andor 0x40, 0x3, var[115 + 5]
-andor 0x40, 0x3, var[115 + 6]
-andor 0x40, 0x0, var[115 + 7]
-andor 0x40, 0x0, var[115 + 8]
 return
+HASCHEL_SLOT0:
# Shift party slots*
+mov var[17][2], stor[24]
+mov var[17][1], var[17][0]
+mov stor[24], var[17][1]
+EMPTY_SLOT2:
+mov 0xffffffff, stor[25]
+mov stor[25], var[17][2]
+jmp_cmp ==, 0xffffffff, var[17][0], inl[:SOLO_HASCHEL]
+jmp inl[:LABEL_152]
+SOLO_HASCHEL:
# Replace solo Haschel with Dart
+mov 0x0, var[17][0]
+jmp inl[:LABEL_152]
 LABEL_153:
 andor 0x40, 0x23, var[115 + 0]
 andor 0x40, 0x0, var[115 + 1]
